File: 01/part1.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin
});

let total = 0;
rl.on('line', mass => {
  const fuel = Math.floor(mass / 3) - 2;
  total += fuel;
});

rl.on('close', () => {
  console.log(`Total: ${total}`);
});

File: 01/part1.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var (
		input string
		total int64
		n     int
	)
	n, _ = fmt.Scanln(&input)
	for n > 0 {
		mass, _ := strconv.ParseInt(input, 10, 64)
		fuel := (mass / 3) - 2
		total += fuel
		n, _ = fmt.Scanln(&input)
	}
	fmt.Println("Total:", total)
}
